### PR TITLE
Merge latest docs from private operator docs

### PR DESCRIPTION
**Pull request by ravenac95**
We are preparing to have these docs merged in for tomorrow. The docs have been reviewed by @david-yan and myself. 

Feel free to suggest any more adjustments

### CODE REVIEW

File: src/operators/joining-the-testnet.md [hunk 1]
```diff
@@ -0,0 +1,542 @@
...
+
+```bash
+$ mkdir -p /serverdir/{etc,node/entity}
+```
+
+#### Copy the Node artifacts from `/localhostdir`
+
+In order for the node registration to work properly, as defined in
+`/localhostdir/entity.json`, you must copy the node's artifacts you generated in
+the [Initializing a Node](https://github.com/) section. To do so upload the
```
Comment (ptrus) [inline src/operators/joining-the-testnet.md]: links to `https://github.com`

File: src/operators/joining-the-testnet.md [hunk 2]
```diff
@@ -0,0 +1,542 @@
...
+
+    # The external_address option is used when registering this node to the
+    # network. If you are using a sentry node then this should be the public
+    # ip of the sentry node.
+    external_address: tcp://{{ external_address }}:26656
+
+  db:
+    backend: boltdb
+  debug:
+    addr_book_lenient: False
```
Comment (ptrus) [inline src/operators/joining-the-testnet.md]: this is default (could be skipped)

File: src/operators/joining-the-testnet.md [hunk 3]
```diff
@@ -0,0 +1,542 @@
...
+  core:
+    listen_address: tcp://0.0.0.0:26656
+
+    # The external_address option is used when registering this node to the
+    # network. If you are using a sentry node then this should be the public
+    # ip of the sentry node.
+    external_address: tcp://{{ external_address }}:26656
+
+  db:
+    backend: boltdb
```
Comment (ptrus) [inline src/operators/joining-the-testnet.md]: this is default (could be skipped)

File: src/operators/joining-the-testnet.md [hunk 4]
```diff
@@ -0,0 +1,542 @@
...
+8e3873f84f7f2250eb456dc598dc56b812f561137fe41c383128e6c14e0e2e74
+cf3b004bd98f3e1eab92e97c5fa6cbe4d42a00133c515a2440fefaca514a48ff
+```
+
+Once you get to a node that's connected you can move on to the next section as
+your node is not yet registered as a validator on the Oasis Testnet.
+
+## Signing up for Testnet Tokens
+
+_This won't be necessary if you are in the genesis file or already have tokens
```
Comment (ptrus) [inline src/operators/joining-the-testnet.md]: > For most people this, will not be true

unneeded (or wrongly placed) comma i think

File: src/operators/joining-the-testnet.md [hunk 5]
```diff
@@ -0,0 +1,542 @@
...
+minimum stake required to register an entity and register a node as a validator
+is 100 tokens. So we will generate an escrow transaction that escrows 100 tokens
+on your own Entity.
+
+```
+$ oasis-node stake account gen_escrow \
+    --entity $ENTITY_DIR_PATH \
+    --stake.escrow.account $ACCOUNT_ID \
+    --stake.transaction.amount 100000000000000000000 \
+    --stake.transaction.file $OUTPUT_TX_FILE_PATH \
```
Comment (ptrus) [inline src/operators/joining-the-testnet.md]: this relies on https://github.com/oasislabs/oasis-core/pull/2374 right? (so should probably not be merged before that pr)

File: src/operators/joining-the-testnet.md [hunk 6]
```diff
@@ -0,0 +1,542 @@
...
+##
+
+# Set this to where you wish to store node data. The node artifacts
+# should also be located in this directory (for us this is /serverdir/node)
+datadir: /serverdir/node
+
+## THESE NEXT 3 LINES ARE TEMPORARY YOU SHOULD NOT EXPOSE THIS PORT IN ANY WAY
+grpc:
+  debug:
+    port: "42261"
```
Comment (ptrus) [inline src/operators/joining-the-testnet.md]: not 100% sure why is this needed (cannot the unix socket be used?)

File: src/operators/joining-the-testnet.md [hunk 7]
```diff
@@ -0,0 +1,542 @@
...
+
+As part of the starting the server process, the `oasis-node` binary will, by
+default, setup an internal unix socket in the `datadir` of the Node. This socket
+can be used to communicate to the node and query details about the network.
+
+```bash
+$ oasis-node registry entity list -a /serverdir/node/internal.sock
+```
+
+If this command fails, you'll receive a non-zero exit code and there's a high
```
Comment (ptrus) [inline src/operators/joining-the-testnet.md]: > not connected to the network

(e.g. node could still be syncing - which can be checked with [`is-synced`](https://github.com/oasislabs/oasis-core/blob/99213538da0b9b602b2b9a962c7c276fbd61ea7c/go/oasis-node/cmd/control/control.go#L26) command

File: src/operators/joining-the-testnet.md [hunk 8]
```diff
@@ -0,0 +1,542 @@
...
+    # The external_address option is used when registering this node to the
+    # network. If you are using a sentry node then this should be the public
+    # ip of the sentry node.
+    external_address: tcp://{{ external_address }}:26656
+
+  db:
+    backend: boltdb
+  debug:
+    addr_book_lenient: False
+  seeds: "{{ seed_node_address }}"
```
Comment (ptrus) [inline src/operators/joining-the-testnet.md]: as of https://github.com/oasislabs/oasis-core/pull/2300 this is a list, so should be:
```
seeds:
  - "{{ seed_node_address }}"
```

File: src/operators/joining-the-testnet.md (hunk 6 above)
Comment (ptrus) [reply src/operators/joining-the-testnet.md]: if it's due to the "Port forwarding" step (at least on linux port forwarding to a unix domain socket should work), otherwise some magic with `socat` might be needed

File: src/operators/architectural-overview.md [hunk 9]
```diff
@@ -7,24 +7,77 @@ is used to provide enough guidance to be useful for the purposes of getting
...
+
+A Committee is a set of Nodes that are participating in the same service layer
+of oasis. Committees are described in more detail in the [Modular
+Architecture](#modular-architecture) Section.
+
 ## Modular Architecture
 
 The Oasis Network uses a modular architecture similar to that of a Service
-Oriented Archiecture or a Microservices Architecture. Any given node
+Oriented Archiecture or a Microservices Architecture. Any given Node
```
Comment (peterjgilbert) [inline src/operators/architectural-overview.md]:
```suggestion
Oriented Architecture or a Microservices Architecture. Any given Node
```

File: src/operators/joining-the-testnet.md (hunk 1 above)
Comment (david-yan) [reply src/operators/joining-the-testnet.md]: fixed

File: src/operators/joining-the-testnet.md (hunk 6 above)
Comment (ravenac95) [reply src/operators/joining-the-testnet.md]: Ya I was mostly worried about a permissions issue on the `ssh`ing user... the port would lead to less questions from people using the doc... was the thought

File: src/operators/architectural-overview.md [hunk 10]
```diff
@@ -7,24 +7,77 @@ is used to provide enough guidance to be useful for the purposes of getting
...
+
+### Node
+
+A Node is a device (VM, Bare Metal, Container, etc.) that is participating in a
+committee on the network. Each Node has a private key that is used for signing
+operations during block production and a public key, or Node Identity, used for
+identification. See [Entities and Key Management](#entities-and-key-management)
+for further information.
+
+### Committee
```
Comment (peterjgilbert) [inline src/operators/architectural-overview.md]:
```suggestion
### Committees
```

File: src/operators/architectural-overview.md [hunk 11]
```diff
@@ -4,27 +4,80 @@ As an operator of a Node on the Oasis Network, it is suggested that you have an
...
+
+A Committee is a set of Nodes that are participating in the same service layer
+of the Oasis Network. Committees are described in more detail in the [Modular
+Architecture](#modular-architecture) section.
 
 ## Modular Architecture
 
 The Oasis Network uses a modular architecture similar to that of a Service
-Oriented Archiecture or a Microservices Architecture. Any given node
+Oriented Archiecture or a Microservices Architecture. Any given Node
```
Comment (peterjgilbert) [inline src/operators/architectural-overview.md]:
```suggestion
Oriented Architecture or a Microservices Architecture. Any given Node
```

File: src/operators/maintenance/wiping-node-state.md [hunk 12]
```diff
@@ -0,0 +1,34 @@
+# Wiping node state
```
Comment (peterjgilbert) [inline src/operators/maintenance/wiping-node-state.md]:
```suggestion
# Wiping Node State
```

File: src/operators/joining-the-testnet.md [hunk 13]
```diff
@@ -0,0 +1,554 @@
...
+use for deployment. These machines are the following:
+
+* Your local system, henceforth called the `localhost`.
+* A remote system to run as an Oasis node, henceforth called the `server`.
+
+The reason for this is to ensure protection of the keys used to
+setup your node. This guide does not include the use of HSMs, but
+should be used in the future.
+
+## Creating your entity
```
Comment (peterjgilbert) [inline src/operators/joining-the-testnet.md]:
```suggestion
## Creating Your Entity
```

File: src/operators/joining-the-testnet.md [hunk 14]
```diff
@@ -0,0 +1,554 @@
...
+
+Everything in this section should be done on the `localhost` as there are
+sensitive items that will be created.
+
+### Creating a working directory
```
Comment (peterjgilbert) [inline src/operators/joining-the-testnet.md]:
```suggestion
### Creating a Working Directory
```

File: src/operators/joining-the-testnet.md [hunk 15]
```diff
@@ -0,0 +1,554 @@
...
+  You should initialize this information on a system with access to the entity's
+  private key. The directory permissions should be `rwx------`
+
+To create the directory structure, use the following command:
+
+```bash
+$ mkdir -m700 -p {entity,node}
+```
+
+### Copy the genesis file
```
Comment (peterjgilbert) [inline src/operators/joining-the-testnet.md]:
```suggestion
### Copy the Genesis File
```

File: src/operators/joining-the-testnet.md [hunk 16]
```diff
@@ -0,0 +1,554 @@
...
+* `identity.pem` - The node's identity private key. **DO NOT SHARE**
+* `identity_pub.pem` - The node's identity public key.
+* `node_genesis.json` - The node's details if you wish to include this node on
+  the genesis file of the network.
+* `p2p.pem` - The node's private key for libp2p. **DO NOT SHARE**
+* `p2p_pub.pem` - The node's public key for libp2p.
+* `tls_identity.pem` - The node's TLS key for grpc's TLS . **DO NOT SHARE**
+* `tls_identity_cert.pem` - The node's TLS cert for grpc's TLS.
+
+#### Add the node to the Entity descriptor
```
Comment (peterjgilbert) [inline src/operators/joining-the-testnet.md]:
```suggestion
#### Adding the Node to the Entity Descriptor
```

File: src/operators/joining-the-testnet.md [hunk 17]
```diff
@@ -0,0 +1,554 @@
...
+```
+$ oasis-node registry entity update \
+  --datadir /localhostdir/entity \
+  --entity.node.descriptor node_genesis.json
+```
+
+This will update the entity descriptor `entity.json` and subsequently the
+`entity_genesis.json` file that contains the signed entity descriptor payload.
+
+#### Initializing additional nodes
```
Comment (peterjgilbert) [inline src/operators/joining-the-testnet.md]:
```suggestion
#### Initializing Additional Nodes
```

File: src/operators/joining-the-testnet.md [hunk 18]
```diff
@@ -0,0 +1,554 @@
...
+other committees (no compute, no key management, no storage, etc.). At this time
+this documentation does not include instructions in configuring anything beyond
+a single validator. If you'd like to create more validator nodes you can simply
+repeat the process above to initialize the artifacts for an additional node (and
+renaming the proper things). Each node will require at least the network's
+defined minimum staking amount (at this time this is 100 tokens).
+
+## Running an Oasis Node on the `server`
+
+### Setting up the Oasis Node's working directory
```
Comment (peterjgilbert) [inline src/operators/joining-the-testnet.md]:
```suggestion
### Setting up the Oasis Node's Working Directory
```

File: src/operators/joining-the-testnet.md [hunk 19]
```diff
@@ -0,0 +1,554 @@
...
+
+Before we run the node on the `server` we need to ensure that we have a place to
+store necessary files for the node. For this guide, we will call this working
+directory `/serverdir` directory.
+
+#### Setting up the the `/serverdir` directory
```
Comment (peterjgilbert) [inline src/operators/joining-the-testnet.md]:
```suggestion
#### Setting up the the `/serverdir` Directory
```

File: src/operators/joining-the-testnet.md [hunk 20]
```diff
@@ -0,0 +1,554 @@
...
+* `node/` - this is to store the node's data
+* `node/entity/` - this is to store the public components of the node's entity
+
+You can make this directory structure by calling the following command:
+
+```bash
+$ mkdir -p /serverdir/{etc,node/entity}
+```
+
+#### Copy the Node artifacts from `/localhostdir`
```
Comment (peterjgilbert) [inline src/operators/joining-the-testnet.md]:
```suggestion
#### Copying the Node Artifacts from `/localhostdir`
```

File: src/operators/joining-the-testnet.md [hunk 21]
```diff
@@ -0,0 +1,554 @@
...
+* `p2p.pem`
+* `p2p_pub.pem`
+* `tls_identity.pem`
+* `tls_identity_cert.pem`
+
+_You may notice that some of these files were listed as **DO NOT SHARE**. In the
+future these keys should be generated and referenced from HSM. Before we have
+HSM support, these keys should be kept as secure as possible on the `server`.
+
+#### Copy the public Entity artifacts from `/localhostdir`
```
Comment (peterjgilbert) [inline src/operators/joining-the-testnet.md]:
```suggestion
#### Copying the Public Entity Artifacts from `/localhostdir`
```

File: src/operators/joining-the-testnet.md [hunk 22]
```diff
@@ -0,0 +1,554 @@
...
+
+We will also need to have public entity artifacts from the `/localhostdir`. Copy
+the file from `/localhostdir/entity/entity.json` to
+`/serverdir/node/entity/entity.json`.
+
+#### Copy the genesis files
```
Comment (peterjgilbert) [inline src/operators/joining-the-testnet.md]:
```suggestion
#### Copying the Genesis Files
```

File: src/operators/joining-the-testnet.md [hunk 23]
```diff
@@ -0,0 +1,554 @@
...
+by running the command below. Please note, the node is, by default, configured to run in the
+foreground. You will need to use a process supervisor like
+[supervisord](http://supervisord.org/),
+[systemd](https://github.com/systemd/systemd), etc.
+
+```
+$ oasis-node --config /serverdir/etc/config.yml
+```
+
+### Verifying the connection to the network
```
Comment (peterjgilbert) [inline src/operators/joining-the-testnet.md]:
```suggestion
### Verifying the Connection to the Network
```

File: src/operators/joining-the-testnet.md [hunk 24]
```diff
@@ -0,0 +1,554 @@
...
+```
+
+## Staking and Registering
+
+_This won't be necessary if your Entity is in the genesis file._
+
+Once you have been funded, you can now complete the process of connecting your
+node by staking so that you can register your entity and register your node.
+
+### Generate a Staking (Escrow) transaction on the `localhost`
```
Comment (peterjgilbert) [inline src/operators/joining-the-testnet.md]:
```suggestion
### Generating a Staking (Escrow) Transaction on the `localhost`
```

File: src/operators/joining-the-testnet.md [hunk 25]
```diff
@@ -0,0 +1,554 @@
...
+
+  account_id_from_b64("<YOUR ENTITY PUBLIC KEY>")
+  >>> "deadbeefdeadbeefdeadbeef..."
+  ```
+
+Note that the option `--stake.transaction.amount` looks like a very large
+number. This is actually equivalent to 100 tokens on the Testnet as each unit
+value used to track the account balance is 1x10^-18 tokens.
+
+### Submitting your transaction on the `server`
```
Comment (peterjgilbert) [inline src/operators/joining-the-testnet.md]:
```suggestion
### Submitting Your Transaction on the `server`
```

File: src/operators/joining-the-testnet.md [hunk 26]
```diff
@@ -0,0 +1,554 @@
...
+  ```
+
+  Without docker
+  ```
+  $ oasis-node stake account submit \
+    --stake.transaction.file /serverdir/signed-escrow.tx \
+    -a /serverdir/node/internal.sock
+  ```
+
+### Registering you entity
```
Comment (peterjgilbert) [inline src/operators/joining-the-testnet.md]:
```suggestion
### Registering Your Entity
```

File: src/operators/joining-the-testnet.md [hunk 27]
```diff
@@ -0,0 +1,554 @@
...
+
+As a final step, you now need to register your entity onto the network.
+
+**TEMPORARY FIX**: At this time many of these instructions are a temporary fix
+due to missing features on the current `oasis-node` cli.
+
+#### Port Forward
```
Comment (peterjgilbert) [inline src/operators/joining-the-testnet.md]:
```suggestion
#### Port Forwarding
```

File: src/operators/joining-the-testnet.md [hunk 28]
```diff
@@ -0,0 +1,554 @@
...
+a port forwarding so you can make entity requests. _Unfortunately, if you're
+using docker on your `localhost` you will also have to ssh to your `server`
+within your the docker container. We apologize for the inconvenience this is a
+temporary measure._
+
+```
+$ ssh $USER@$SERVER_IP -L 42261:127.0.0.1:42261 -N
+```
+
+#### Register your node onto the network
```
Comment (peterjgilbert) [inline src/operators/joining-the-testnet.md]:
```suggestion
#### Register Your Node on the Network
```

File: src/operators/joining-the-testnet.md [hunk 29]
```diff
@@ -0,0 +1,554 @@
...
+
+On a different terminal from your ssh port forward above, execute the following:
+
+```
+$ oasis-node registry entity register --datadir /localhostdir/entity
+```
+
+If everything is successful you will get a zero exit code.
+
+### Checking that your node is properly registered
```
Comment (peterjgilbert) [inline src/operators/joining-the-testnet.md]:
```suggestion
### Checking that Your Node is Properly Registered
```

File: src/operators/joining-the-testnet.md [hunk 30]
```diff
@@ -0,0 +1,554 @@
...
+
+_We understand these instructions for verification need to improve. Any
+assistance is welcome ;)_
+
+To ensure that your node is properly connected as a Validator on the network, We
+can check to see the node's identity on in the registry's node list.
+Unfortunately, at this time this is a bit of a manual process.
+
+### Get the node's consensus_pub.pem identity
```
Comment (peterjgilbert) [inline src/operators/joining-the-testnet.md]:
```suggestion
### Getting the Node's consensus_pub.pem Identity
```

File: src/operators/joining-the-testnet.md [hunk 31]
```diff
@@ -0,0 +1,554 @@
...
+
+```
+-----BEGIN ED25519 PUBLIC KEY-----
+s+vZ71qeZnlq0HmQSDBiWn2OKcy3fXOuPMu76/5GkUI=
+-----END ED25519 PUBLIC KEY-----
+```
+
+We will use `s+vZ71qeZnlq0HmQSDBiWn2OKcy3fXOuPMu76/5GkUI=`. As the key to search for.
+
+### grep for the node's identity
```
Comment (peterjgilbert) [inline src/operators/joining-the-testnet.md]:
```suggestion
### grepping for the Node's Identity
```

File: src/operators/joining-the-testnet.md [hunk 32]
```diff
@@ -0,0 +1,554 @@
+# Joining the Testnet
+
+This guide will cover setting up your nodes on the public testnet. There is some
```
Comment (peterjgilbert) [inline src/operators/joining-the-testnet.md]:
```suggestion
This guide will cover setting up your nodes on the Public Testnet. There is some
```

File: src/operators/joining-the-testnet.md [hunk 33]
```diff
@@ -0,0 +1,554 @@
...
+
+### Initializing an Entity
+
+The entity, as [described
+here](../architectural-overview.md#entities-and-key-management) is critical to
+operating nodes on the network as it controls the stake attached to a given
+individual or organization on the network. In the future we will support using
+entity keys through HSMs to ensure that entity keys cannot be easily
+compromised. We strongly suggest that you do not use any entity that is generated
+with the current process on the Mainnet. During the public testnet and staking
```
Comment (peterjgilbert) [inline src/operators/joining-the-testnet.md]:
```suggestion
with the current process on the Mainnet. During the Public Testnet and staking
```

File: src/operators/joining-the-testnet.md [hunk 34]
```diff
@@ -0,0 +1,554 @@
...
+$ export NODE_PUB_KEY="s+vZ71qeZnlq0HmQSDBiWn2OKcy3fXOuPMu76/5GkUI="
+$ oasis-node registry node list -v -a /serverdir/node/internal.sock | grep $NODE_PUB_KEY
+```
+
+If `grep` found the key, then you're properly connected!
+
+## You're a Validator!
+
+If you've made it this far you've properly connected your node to the network
+and you're now a Validator on the public testnet.
```
Comment (peterjgilbert) [inline src/operators/joining-the-testnet.md]:
```suggestion
and you're now a Validator on the Public Testnet.
```

File: src/operators/architectural-overview.md [hunk 35]
```diff
@@ -4,27 +4,80 @@ As an operator of a Node on the Oasis Network, it is suggested that you have an
...
+be scheduled are chosen from the Registry.
+
+It is possible that a Node can register for more than one committee to make
+efficient use of hardware. If, for instance, a Node registers with the
+possibility of operating as a Storage or Compute Node, the Committee Scheduler
+will assign it a single job during a given lifetime.
+
+#### Consensus Committee
+
+The Consensus Committee is the BFT consensus layer. It's primary function during
```
Comment (peterjgilbert) [inline src/operators/architectural-overview.md]:
```suggestion
The Consensus Committee is the BFT consensus layer. Its primary function during
```

File: src/operators/architectural-overview.md [hunk 36]
```diff
@@ -4,27 +4,80 @@ As an operator of a Node on the Oasis Network, it is suggested that you have an
...
+Architecture](#modular-architecture) section.
 
 ## Modular Architecture
 
 The Oasis Network uses a modular architecture similar to that of a Service
-Oriented Archiecture or a Microservices Architecture. Any given node
+Oriented Architecture or a Microservices Architecture. Any given Node
 participates in one of many different committees. These committees each have
 different responsibilities in the execution of smart contracts on the Oasis
 Network. The responsibilities of these committees can be mapped to a set of 4
```
Comment (peterjgilbert) [inline src/operators/architectural-overview.md]:
```suggestion
Network. The responsibilities of these committees can be mapped to a set of four
```

File: src/operators/architectural-overview.md [hunk 37]
```diff
@@ -4,27 +4,80 @@ As an operator of a Node on the Oasis Network, it is suggested that you have an
...
+#### Consensus Committee
+
+The Consensus Committee is the BFT consensus layer. It's primary function during
+transaction processing is to reach consensus on the final state of the
+application that it receives from the Compute layer.
 
 ## Networking Protocols
 
-The Oasis Network uses 3 main protocols for communication:
+The Oasis Network uses 3 different protocols for communication:
```
Comment (peterjgilbert) [inline src/operators/architectural-overview.md]:
```suggestion
The Oasis Network uses three different protocols for communication:
```

File: src/operators/joining-the-testnet.md [hunk 38]
```diff
@@ -0,0 +1,554 @@
...
+these tokens to register your entity and stake on the network. To get tokens,
+you'll need to sign up with [this
+form](https://oasisfoundation.typeform.com/to/dlcekq). While filling out the
+form, it will ask for your Entity Public Key. This is the same as your Entity
+Account ID and can be found in the `/localhostdir/entity/entity.json`. In this
+JSON file, the Entity Public Key is the `id` field.
+
+### Example
+
+In the following example `entity.json`, the Entity Public key is
```
Comment (peterjgilbert) [inline src/operators/joining-the-testnet.md]:
```suggestion
In the following example `entity.json`, the Entity Public Key is
```

File: src/operators/joining-the-testnet.md [hunk 39]
```diff
@@ -0,0 +1,554 @@
...
+  If you're using docker use:
+
+  ```
+  $ docker exec -it oasis_node /bin/bash
+  $ oasis-node stake account submit \
+    --stake.transaction.file /serverdir/signed-escrow.tx \
+    -a /serverdir/node/internal.sock
+  ```
+
+  Without docker
```
Comment (peterjgilbert) [inline src/operators/joining-the-testnet.md]:
```suggestion
  Without docker:
  
```

File: src/operators/joining-the-testnet.md (hunk 5 above)
Comment (peterjgilbert) [reply src/operators/joining-the-testnet.md]: @ravenac95 ?
